manager/manager.go

```diff
@@ -1237,6 +1237,24 @@ func (m *manager) watchForNewOoms() error {
 			if err != nil {
 				klog.Errorf("failed to add OOM kill event for %q: %v", oomInstance.ContainerName, err)
 			}
+
+			// Count OOM events for later collection by prometheus
+			request := v2.RequestOptions{
+				IdType: v2.TypeName,
+				Count:  1,
+			}
+			conts, err := m.getRequestedContainers(oomInstance.ContainerName, request)
+			if err != nil {
+				klog.Errorf("failed getting container info for %q: %v", oomInstance.ContainerName, err)
+				continue
+			}
+			if len(conts) != 1 {
+				klog.Errorf("Expected the request to match only one container")
+				continue
+			}
+			for _, cont := range conts {
+				cont.oomEvents++
```

Comment:
[It is not safe](https://golang.org/ref/mem) to increment this counter from one goroutine while another might read from it.

Reply:
What change do you suggest? Replace the increment with `atomic.AddUint64`?

Reply:
Or using a lock. Keep in mind that no matter what you do you will have to serialize reads too.

Reply:
Moved it to atomics